### ISSUE TITLE
Discrete Flow: Likelihood per sample

### ISSUE THREAD

**Issue by yannadani** [google/edward2#324]
I was wondering how to get likelihood per sample in the discrete flow case. 

I checked out the implementation of quantized rings of gaussian experiment and found that for training the model, the flow samples are used as labels and the base distribution as logits. I thought we are supposed to compute NLL on the model likelihood rather than samples.
Maybe I am slightly confused wrt implementation. So any clarification would be great!
